docs/specification/draft/client/elicitation.mdx

```diff
@@ -151,139 +257,236 @@ To request information from a user, servers send an `elicitation/create` request
+
+<Note>
+  **Important**: Out-of-band elicitation is *not* for re-authorizing the MCP
+  client's access to the MCP server (that is covered by [MCP
+  authorization](../basic/authorization)). It's specifically for scenarios where
+  the MCP server needs to obtain sensitive information, or authorization for
+  third-party services on behalf of the user. The MCP client's authorization
+  (bearer) token remains unchanged throughout this process.
+</Note>
+
+Out-of-band elicitation requests **MUST** specify `mode: "oob"` and include these parameters:
+
+| Name            | Type   | Description                               |
+| --------------- | ------ | ----------------------------------------- |
+| `url`           | string | The URL that the user should navigate to. |
+| `elicitationId` | string | A unique identifier for the elicitation.  |
+
+The `url` parameter **MUST** contain a valid URL. The `message` parameter **MUST NOT** contain a URL.
+
+#### Example: Request Sensitive Data
+
+This example shows an out-of-band elicitation request directing the user to a secure URL where they can provide sensitive information (an API key, for example).
+The same request could direct the user into an OAuth authorization flow, or a payment flow. The only difference is the URL and the message.
+
+**Request:**
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 3,
+  "method": "elicitation/create",
+  "params": {
+    "mode": "oob",
+    "elicitationId": "550e8400-e29b-41d4-a716-446655440000",
+    "url": "https://mcp.example.com/ui/set_api_key",
+    "message": "Please provide your API key to continue."
+  }
+}
+```
+
+**Response:**
 
 ```json
 {
   "jsonrpc": "2.0",
-  "id": 2,
+  "id": 3,
   "result": {
-    "action": "reject"
+    "action": "accept"
+  }
+}
+```
+
+The response with `action: "accept"` indicates that the user has consented to the
+interaction. It does not mean that the interaction is complete. The interaction occurs out
+of band and the client is not aware of the outcome, unless the client requests progress updates.
+
+### Progress Tracking
+
+The client **MAY** request progress updates from the server by sending an `elicitation/track` request with a [progress token](/specification/draft/basic/utilities/progress#progress-token). This is particularly useful in out-of-band mode, because the client is not involved in the interaction.
+
+The client **MUST** include an `elicitationId` in the request to identify which elicitation to send progress updates for. The client **SHOULD** include a `progressToken` in the request's `_meta` field.
+
+The server **MUST** ignore any `elicitation/track` requests containing an `elicitationId` that is not known or does not belong to the client.
+
+The server **MAY** send a `notifications/progress` notification to the client with the progress token and the progress status.
+
+The server **SHOULD** send a `elicitation/track` response to the client once the elicitation is complete.
```

Comment:
I think this is the first time we have an MCP request that could take a long time (minutes+) to complete. This may be a concern for implementors who use Streamable HTTP and are subject to timeouts are various network layers. May consider recommending https://github.com/modelcontextprotocol/modelcontextprotocol/pull/899 as a way to handle them.

Reply:
tools/call requests can take minutes, no?